packages/apps/reputation-oracle/server/src/modules/webhook/webhook.service.ts

```diff
@@ -40,47 +57,160 @@ export class WebhookService {
+    webhookEntity.waitUntil = new Date();
+    webhookEntity.retriesCount = 0;
+
+    webhookEntity =
+      await this.webhookIncomingRepository.createUnique(webhookEntity);
+
+    if (!webhookEntity) {
+      throw new ControlledError(ErrorWebhook.NotCreated, HttpStatus.NOT_FOUND);
+    }
+  }
+
+  /**
+   * Creates an outgoing webhook entry in the repository.
+   * Sets initial status to 'PENDING' and stores the provided payload, hash, and URL.
+   * @param {object} payload - The payload to send in the webhook.
+   * @param {string} hash - A hash generated from the URL and payload for unique identification.
+   * @param {string} url - The destination URL for the outgoing webhook.
+   * @throws {ControlledError} If the webhook cannot be created.
+   */
+  public async createOutgoingWebhook(
+    payload: object,
+    hash: string,
+    url: string,
+  ): Promise<void> {
+    let webhookEntity = new WebhookOutgoingEntity();
+    webhookEntity.payload = payload;
+    webhookEntity.hash = hash;
+    webhookEntity.url = url;
+    webhookEntity.status = WebhookOutgoingStatus.PENDING;
     webhookEntity.waitUntil = new Date();
     webhookEntity.retriesCount = 0;
 
-    webhookEntity = await this.webhookRepository.createUnique(webhookEntity);
+    webhookEntity =
+      await this.webhookOutgoingRepository.createUnique(webhookEntity);
 
     if (!webhookEntity) {
       throw new ControlledError(ErrorWebhook.NotCreated, HttpStatus.NOT_FOUND);
     }
   }
 
   /**
-   * Handles errors that occur during webhook processing.
-   * It logs the error and, based on retry count, updates the webhook status accordingly.
-   * @param webhookEntity - The entity representing the webhook data.
-   * @param error - The error object thrown during processing.
-   * @returns {Promise<void>} - Returns a promise that resolves when the operation is complete.
+   * Creates a tracking record for escrow completion in the repository.
+   * Sets initial status to 'PENDING'.
+   * @param {ChainId} chainId - The blockchain chain ID.
+   * @param {string} escrowAddress - The address of the escrow contract.
+   * @throws {ControlledError} If the tracking record cannot be created.
    */
-  public async handleWebhookError(
+  public async createEscrowCompletionTracking(
+    chainId: ChainId,
+    escrowAddress: string,
+  ): Promise<void> {
+    let escrowCompletionTrackingEntity = new EscrowCompletionTrackingEntity();
+    escrowCompletionTrackingEntity.chainId = chainId;
+    escrowCompletionTrackingEntity.escrowAddress = escrowAddress;
+    escrowCompletionTrackingEntity.status =
+      EscrowCompletionTrackingStatus.PENDING;
+    escrowCompletionTrackingEntity.waitUntil = new Date();
+    escrowCompletionTrackingEntity.retriesCount = 0;
+
+    escrowCompletionTrackingEntity =
+      await this.escrowCompletionTrackingRepository.createUnique(
+        escrowCompletionTrackingEntity,
+      );
+
+    if (!escrowCompletionTrackingEntity) {
```

Comment:
Since we have [`handleQueryFailedError(error: QueryFailedError)`](https://github.com/humanprotocol/human-protocol/blob/be1435df2a0a5ab685b89a2559ea97e75e1a0b73/packages/apps/reputation-oracle/server/src/common/errors/database.ts#L11), this handler will throw a database error when it occurs during `createUnique` call. I think we can remove `if (!escrowCompletionTrackingEntity) {...}` that check, when duplicate error occurs, we will log it and complete the webhook.

Reply:
Yup, seems we can remove this check and error will be handler on the higher level